### ISSUE TITLE
High CPU Usage with OpenVINO YOLOv8n on Integrated GPU – How Can I Reduce It?

### ISSUE THREAD

**Issue by AlaaArboun** [ultralytics/ultralytics#19781]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi everyone,

I'm running inference using an OpenVINO-converted YOLOv8n model on an integrated GPU (IGPU), but I'm noticing that the CPU usage stays around 90% while the GPU is only at about 50%. I’ve tried configuring various GPU-specific properties to reduce the CPU load, yet the high CPU usage persists.

```python
import collections
import time
import openvino as ov
import openvino.properties as properties
import openvino.properties.device as device
import openvino.properties.hint as hints
import openvino.properties.streams as streams
import openvino.properties.intel_auto as intel_auto
import cv2
from ultralytics import YOLO
import torch


def open_video_stream():
    return cv2.VideoCapture(0)

model_path = r"\yolov8n_openvino_model\yolov8n.xml"

core = ov.Core()

# Read the quantized model
print("Loading OpenVINO model...")
ov_model = core.read_model(str(model_path))

# Reshape the input for GPU
ov_model.reshape({0: [1, 3, 640, 640]})

gpu_config = {
    
    hints.inference_precision: "FP16",  # Alternatively, use ov.Type.f16 if available in your API
    hints.execution_mode: "PERFORMANCE",
    "ENABLE_CPU_PINNING": "NO",
    "NUM_STREAMS": "1",
    "ENABLE_CPU_PINNING": "NO",
    "COMPILATION_NUM_THREADS": "2",
    "GPU_DISABLE_WINOGRAD_CONVOLUTION": "YES",
    "GPU_QUEUE_THROTTLE": hints.Priority.LOW,
    "GPU_HOST_TASK_PRIORITY": hints.Priority.LOW,
 
}


# Compile the model for GPU
print(f"Compiling model for {device}...")
compiled_model=core.compile_model(ov_model,"GPU",gpu_config)      

det_model = YOLO("yolov8n.pt")


label_map = det_model.model.names  # Extract class names

test_img_path = "coco_bike.jpg"

# Test inference
try:
    test_results = det_model(test_img_path)
    print(f"Test inference successful! Found {len(test_results[0].boxes)} objects")
except Exception as e:
    print(f"Warning: Test inference failed: {e}")
    print("Error details:", e)
    print("Continuing anyway...")

def infer(*args):
    result = compiled_model(args)
    return torch.from_numpy(result[0])


det_model.predictor.inference = infer
det_model.predictor.model.pt = False   # Indicate PyTorch model is not used




def run_object_detection():
    print("Opening video stream...")
    cap = open_video_stream()
    
    if not cap.isOpened():
        print("Error: Could not open RTSP stream.")
        return
    
    print("Starting object detection loop...")
    processing_times = collections.deque()
    
    while True:
        ret, frame = cap.read()
        if not ret:
            print("Failed to get frame from stream. Retrying...")
            # Try to reopen the stream if it's dropped
            cap.release()
            time.sleep(1)  # Wait a bit before reconnecting
            cap = open_video_stream()
            continue
        frame= cv2.cvtColor(frame, cv2.COLOR_YUV2BGR_NV12)
        # Optionally, resize frame for faster processing if it's too large
        scale = 1280 / max(frame.shape)
        if scale < 1:
            frame = cv2.resize(frame, None, fx=scale, fy=scale, interpolation=cv2.INTER_AREA)
        
      
        try:
            # Run inference on the frame
            results = det_model(frame, verbose=False)
           
            
           
            if len(processing_times) > 200:
                processing_times.popleft()
            
            # Overlay inference time and FPS on the output frame
            output_frame = results[0].plot()
            
            cv2.imshow("annotated frame", output_frame)
            
        except Exception as e:
            print(f"Error during inference: {e}")
            # Show the original frame if inference fails
            cv2.putText(frame, "Inference Error", (20, 40), 
                       cv2.FONT_HERSHEY_COMPLEX, 1, (0, 0, 255), 2, cv2.LINE_AA)
            cv2.imshow("annotated frame", frame)
        
        if cv2.waitKey(1) == 27:  # Exit if ESC key is pressed
            break
    
    print("Cleaning up...")
    cap.release()
    cv2.destroyAllWindows()

if __name__ == "__main__":
    print("Starting application...")
    run_object_detection()
 ```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @AlaaArboun, thank you for reaching out and for providing such a detailed description of your issue! 🚀

We recommend checking out our [Docs](https://docs.ultralytics.com/) for additional information on YOLO usage and configurations. You may find helpful resources like [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples, as well as tips for optimizing your setup.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it effectively. For instance, a simplified script that demonstrates the high CPU usage issue would be highly valuable.

If this is a custom ❓ Question or setup inquiry, ensure you’ve included relevant details such as your environment specs (e.g., Python version, OpenVINO version, hardware details) and any logs or observations that might help us better understand the problem.

For real-time discussions, consider joining our [Discord community](https://discord.com/invite/ultralytics) 🎧. You can also explore [Discourse](https://community.ultralytics.com/) for more in-depth conversations or visit our [Subreddit](https://reddit.com/r/Ultralytics) for community insights and discussions.

## Upgrade

First, ensure you’re using the latest version of the `ultralytics` package. Upgrade with the following command in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

This ensures you benefit from the latest improvements and fixes.

## Environments

YOLO models can be run in various verified environments. Here are a few options for quick experimentation:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

Lastly, you can check the status of our [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests here: <a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>. If this badge is green, all tests are passing and the latest code has been verified.

This is an automated response to guide you in the right direction. An Ultralytics engineer will review your question and provide further assistance soon. Thank you for your patience and for being part of the Ultralytics community! 😊

**Comment by Y-T-G** [comment]
I don't get why you are not loadingvthe OpenVINO model directly in Ultralytics.
